### ISSUE TITLE
Road Proxy/LOD Textures Stop Working If Save Game Was Reloaded

### ISSUE THREAD

**Issue by NeHooYeY** [WolvenKit/WolvenKit#1798]
Description:
Road proxy meshes that have "engine\materials\metal_base_proxy.mt" in preloadLocalMaterialInstances with BaseColor texture shows replaced texture correctly when loading save game from Main Menu, but when player reloads save from pause menu during active game session all  custom proxy textures stop working globally. If player returns to Main Menu and loads save game without game restart, proxy textures working again. 
- Loading game from Main Menu: **Replaced Proxy Textures Work**
- Loading game from Pause Menu: **Replaced Proxy Textures revert to vanilla**
- Return to Main Menu **without game restart** and loading save from there makes **Replaced Proxy Textures work again**.
--------------------------------------------------------------------------------------
Reloading Archives using RHT, then reloading save from Pause Menu(ESC) **does not fix the issue**, also game doesnt crash when doing so if it matters.
There are no "embedded" textures in proxy meshes, paths to textures within mesh have "Default" flag.
Switching from **preloadLocalMaterialInstances** to > **localMaterialBuffer** as a test **does not fix the issue**.
Adding **.streamingsector_inplace** that contains original proxy mesh as well as texture with "embedded" flag, then nulling it completely and testing **does not fix the issue**.
Im adding only proxy textures for roads into my project/.archive, **no .streamingsector files**, as well as no proxy meshes(.prx)
Adding proxy road mesh and custom pathing texture **does not fix the issue**.

- Textures imported using correct vanilla settings from CBitmapTextue > "setup" as .png. 
- I'm not using any LOD altering mods, that could affect my textures.

**To Reproduce**
1. Install "brightE3r_roads.archive" : https://drive.google.com/file/d/1bR6eR3sr7waJEvAAKA4X6vFkdk5W_o9V/view?usp=sharing

2. Go to this location:

(<a href="https://ibb.co/X5fnsZt"><img src="https://i.ibb.co/DMPBCDQ/map.png" alt="map" border="0"></a>)

---------------------------------------------------------------------------------------
3. **Save your game at that spot and either return to Main Menu, or Restart Game**.
---------------------------------------------------------------------------------------
4. **Load That Save Game From Main Menu** and look in Kompeki Plaza direction, you should see working proxy texture:

<a href="https://imgbb.com/"><img src="https://i.ibb.co/x590yn2/works.png" alt="works" border="0"></a> 

5. **Press ESC to enter Pause Menu** and **Reload Same Save**, when game is loaded, you will see that proxy texture reverted to vanilla:

<a href="https://imgbb.com/"><img src="https://i.ibb.co/tmfNR1n/doesnt.png" alt="doesnt" border="0"></a>

**Expected behavior**
Working proxy textures when game was loaded from Main Menu or Pause Menu during active game session after initial save load.

**Version:**
 - WolvenKit Nightly 8.14.1 from June 11, 2024

**Comment by seberoth** [comment]
Don't think that a bug in WolvenKit. If it were, it would never work. But since its working on a "full" reload...
The archive contents won't magicly change on reload.
Probably more like the engine loads another mesh/texture on "soft" reload.

Also if you want that one of us (or another modder) take a look at it, please provide a minimal working example (files/info needed to replace the (proxy) texture in the first place).
Since we won't ( at least not me ) start to fiddle around hundred of files (or how sectors work) just to test something^^
Working on WolvenKit doesn't mean/imply that we know how everything works ingame :)

**Comment by Simarilius-uk** [comment]
they've linked the mod in the steps to reproduce?

**Comment by seberoth** [comment]
I know. And its good to reproduce it ingame.
But at least me doesn't know enough about streamingsectors that I can tell which of the 130 mesh/texture files in the archive is responsible for the street texture at this location (or which proxy is used to load them).

If it is enough for you, feel free to check it.

**Comment by NeHooYeY** [comment]
Oh, yeah, that's confusing.

If it still matters im linking relevant .streamingsector, proxy mesh and proxy textures from screenshots example.

Link: https://drive.google.com/file/d/1OEajZLTwHxF0IbGV2JJUJN75sjsfuLzj/view?usp=sharing

- Path to sector: **base\worlds\03_night_city\_compiled\default\exterior_-2_2_0_4.streamingsector**
- Node: **707**
- Path to prx1.mesh from screenshots: **base\worlds\03_night_city\sectors\_external\road_meshes\r_roadsplinenode_103_ff74f81a\prx1.mesh**


<a href="https://imgbb.com/"><img src="https://i.ibb.co/wYnqVvj/Cyberpunk-2077-C-2020-by-CD-Projekt-RED-18-06-2024-18-14-46.png" alt="Cyberpunk-2077-C-2020-by-CD-Projekt-RED-18-06-2024-18-14-46" border="0"></a>

I see there is indeed "Soft" flag in node mesh DepotPath. I dont know what that means to be honest.

**Comment by seberoth** [comment]
Thanks for the info :)

Didn't test it yet, but node 707 refers to `base\worlds\03_night_city\_compiled\default\4e5438109469f1cf.streamingsector_inplace` which indeed has an embedded version of `base\worlds\03_night_city\sectors\_external\proxy\2727358387\rt_2x2_with_sidewalks_12m_cracks_d.xbm` and `base\worlds\03_night_city\sectors\_external\proxy\3059599730\rt_2x2_with_sidewalks_24m_cracks_d.xbm`

**Comment by NeHooYeY** [comment]
Yes, i've added that .streamingsector_inplace to the project for tests and removed all data from it, no change.
What i had not yet tried is converting it to .json, change all flags to "Default" and convert back from .json

**Comment by seberoth** [comment]
Done some "research" (Modified each file which has `rt_2x2_with_sidewalks_24m_cracks_d.xbm` as embedded and removed it...)...

As soon as I remove it from `base\worlds\03_night_city\_compiled\default\ac5dc8a397fc4856.streamingsector_inplace`, the texture will stay between "ingame" reloads.
Its strange since its only referenced in `base\worlds\03_night_city\_compiled\default\exterior_-3_3_0_4.streamingsector` which doesn't seem to be related...
![grafik](https://github.com/WolvenKit/WolvenKit/assets/456881/b436d19c-58eb-4980-ad85-77e7b0d03e6b)

**Comment by seberoth** [comment]
Maybe `exterior_-3_3_0_4` gets loaded first if reloaded ingame, the texture gets cached and then `exterior_-2_2_0_4` gets loaded and uses that cache.
Not sure why the behaviour is different when someone loads from the main menu

**Comment by NeHooYeY** [comment]
Oh nice, so after removing the texture from the .streamingsector_inplace files, it becomes stable?
